/*
      Dopo aver modificato il .proto file:
          python3 -m grpc_tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/sharedpw.proto

      Questo genera
          - login_pb2.py che contiene le classi di richiesta e risposta
          - login_pb2_grpc.py che contiene le classi client e server
*/

syntax = "proto3";

option java_multiple_files = true;
option java_package = "/SDCC/services";
option java_outer_classname = "SharedPwProto";
option objc_class_prefix = "SHP";
package sharedpw;

service Shared {
  rpc passwordRequest (SharedPasswordRequest) returns (SharedPasswordReply) {}
  rpc checkPassword (CheckSharedPasswordRequest) returns (CheckSharedPasswordReply) {}
}

message SharedPasswordRequest {
 string group_name = 1;
 string email = 2;
 string service = 3;
}

message SharedPasswordReply {
  bool exists = 1;
  string password = 2;
}

message CheckSharedPasswordRequest {
  string group_name = 1;
  string agency = 2;
  string email = 3;
  string password = 4;
}

message CheckSharedPasswordReply {
  bool isChecked = 1;
}